compiler/tests/data/string-singleton.bal

```diff
@@ -14,8 +14,8 @@
 // Z<:XYZ
 // Z<:XZ
 type X "x";
-type Y "y";
-type Z "z";
+type Y "y-y";
+type Z "z-z";
```

Comment:
Need a test with all four of string singleton, char singleton, string type, char type, but I would add a new one and leave this one as is.

Reply:
Added compiler/tests/data/string-all-subtypes.bal